packages/dds/matrix/src/permutationvector.ts

```diff
@@ -378,22 +384,25 @@ export class PermutationVector extends Client {
 				break;
 			}
 
-			default:
+			default: {
 				throw new Error("Unhandled MergeTreeDeltaType");
+			}
 		}
 	};
 
-	private readonly onMaintenance = (args: IMergeTreeMaintenanceCallbackArgs) => {
+	private readonly onMaintenance = (args: IMergeTreeMaintenanceCallbackArgs): void => {
 		if (args.operation === MergeTreeMaintenanceType.UNLINK) {
 			let freed: number[] = [];
 
 			for (const { segment } of args.deltaSegments) {
 				const asPerm = segment as PermutationSegment;
 				if (isHandleValid(asPerm.start)) {
-					// Note: Using the spread operator with `.splice()` can exhaust the stack.
```

Comment:
Given this comment, I wonder if we want to revert the change here and just disable the rule.

Reply:
I was wondering this too, but since we don't use splice anymore, I didn't know if the comment was still relevant or not. I can definitely revert and disable though if we want.

Reply:
Just to be safe, that might be best.